tests/common/test_cgroups.py

```diff
@@ -290,44 +297,928 @@ def test_limits(self):
+
+                expected_cpu_limit = CGroupsLimits.get_default_cpu_limits(cgroup_name)
+                expected_memory_limit = total_ram * 1  # min of %age of totalram vs max gives total ram edge here.
+
+                limits = CGroupsLimits(cgroup_name, resource_configuration=resource_config)
+                self.assertEqual(limits.cpu_limit, expected_cpu_limit)
+                self.assertEqual(limits.memory_limit, expected_memory_limit)
+                self.assertEqual(limits.memory_flags, expected_memory_flags)
+
+                # Configuration
+                # RAM - 1024 MB
+                # # of cores - 8
+                total_ram = 1024
+                patch_get_processor_cores.return_value = 8
+                patch_get_total_mem.return_value = total_ram
+
+                expected_cpu_limit = CGroupsLimits.get_default_cpu_limits(cgroup_name)
+                expected_memory_limit = 256  # It asked for 10% of 1024, which is lower than default, thus you get
+                # default.
+
+                limits = CGroupsLimits(cgroup_name, resource_configuration=resource_config)
+                self.assertEqual(limits.cpu_limit, expected_cpu_limit)
+                self.assertEqual(limits.memory_limit, expected_memory_limit)
+                self.assertEqual(limits.memory_flags, expected_memory_flags)
+
+        data = '''{
+                      "name": "ExampleHandlerLinux",
+                      "version": 1.0,
+                      "handlerConfiguration": {
+                        "linux": {
+                          "resources": {
+                            "memory": {
+                              "max_limit_percentage": 10,
+                              "max_limit_MBs": 2500,
+                              "memory_oom_kill": "enabled"
+                            }
+                          }
+                        },
+                        "windows": {}
+                      }
+                    }
+                    '''
+        handler_config = HandlerConfiguration(json.loads(data))
+        resource_config = handler_config.get_resource_configurations()
+        cgroup_name = "test_cgroup"
+        expected_memory_flags = {"memory_pressure_warning": None, "memory_oom_kill": "enabled"}
+
+        with patch("azurelinuxagent.common.osutil.default.DefaultOSUtil.get_processor_cores") as \
+                patch_get_processor_cores:
+            with patch("azurelinuxagent.common.osutil.default.DefaultOSUtil.get_total_mem") as patch_get_total_mem:
+                # Configuration
+                # RAM - 512 MB
+                # # of cores - 8
+                total_ram = 256
+                patch_get_processor_cores.return_value = 8
+                patch_get_total_mem.return_value = total_ram
+
+                expected_cpu_limit = CGroupsLimits.get_default_cpu_limits(cgroup_name)
+                expected_memory_limit = total_ram * 1  # min of %age of totalram vs max gives total ram edge here.
```

Comment:
Why do we allow setting the memory limit for extensions to something higher than the maximal allowed limit for extensions?

Reply:
Can you explain this comment?